src/sagas/storageDomains.js

```diff
@@ -20,111 +20,119 @@ import {
-  // so why not to store them when we have them fresh
-  const dataCentersInternal = dataCentersApi.data_center.map(dataCenter => Api.dataCenterToInternal({ dataCenter }))
-  yield put(setDataCenters(dataCentersInternal))
-
-  // transform all storage domains from dataCentersApi.data_center[*].storage_domains.storage_domain[*]
-  const storageDomainsInternal = []
-  for (const dataCenter of dataCentersApi.data_center) {
-    const storageDomains = dataCenter.storage_domains && dataCenter.storage_domains.storage_domain
-    if (storageDomains) {
-      storageDomainsInternal.push(
-        ...(
-          yield all(
-            storageDomains.map(
-              function* (storageDomain) {
-                let storageDomainInternal = Api.storageDomainToInternal({ storageDomain })
-                storageDomainInternal.permits = yield fetchPermits({ entityType: PermissionsType.STORAGE_DOMAIN_TYPE, id: storageDomain.id })
-                storageDomainInternal.canUserUseDomain = canUserUseStorageDomain(storageDomainInternal.permits)
-                return storageDomainInternal
-              }
-            )
-          )
-        )
-      )
+  yield put(setDataCenters(dataCenters))
+  yield put(setStorageDomains(storageDomains))
+}
+
+function* fetchDataCenters () {
+  const payload = { additional: [ 'permissions', 'storage_domains' ] }
+  const dataCentersApi = yield callExternalAction('getAllDataCenters', Api.getAllDataCenters, { payload })
+
+  if (dataCentersApi && dataCentersApi.data_center) {
+    const dataCentersInternal = dataCentersApi.data_center.map(
+      dataCenter => Transforms.DataCenter.toInternal({ dataCenter })
+    )
+
+    // Calculate permits and 'canUser*'
+    for (const dataCenter of dataCentersInternal) {
+      dataCenter.userPermits = yield permissionsToUserPermits(dataCenter.permissions)
     }
+
+    return dataCentersInternal
   }
 
-  // since storage domains can be attached to >1 data center, collate the copies
-  // and track the domain's status per data center
-  const storageDomainsMerged = mergeStorageDomains(storageDomainsInternal)
-  yield put(setStorageDomains(storageDomainsMerged))
+  return []
 }
 
-/**
- * From a list of storage domains, where a storage domain may occur multiple times
- * with a different data center, build a unique list of storage domains.  Each storage
- * domain returned will contain a data center to SD status map.
- */
-function mergeStorageDomains (storageDomainsInternal) {
-  const idToStorageDomain = storageDomainsInternal.reduce(
-    (accum, storageDomain) => {
-      const merged = accum[storageDomain.id]
-      if (merged) {
-        Object.assign(merged.statusPerDataCenter, storageDomain.statusPerDataCenter)
-      } else {
-        accum[storageDomain.id] = storageDomain
-      }
-      return accum
-    },
-    {}
-  )
+function* fetchDataAndIsoStorageDomains () {
+  const storageDomainsApi = yield callExternalAction('getStorages', Api.getStorages, { payload: {} })
```

Comment:
`payload` is not necessary too

Reply:
I opened #1059 to cover any of these kinds of changes.